src/net/sourceforge/kolmafia/session/LocketManager.java

```diff
@@ -0,0 +1,111 @@
+package net.sourceforge.kolmafia.session;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.Modifiers;
+import net.sourceforge.kolmafia.MonsterData;
+import net.sourceforge.kolmafia.objectpool.ItemPool;
+import net.sourceforge.kolmafia.persistence.DebugDatabase;
+import net.sourceforge.kolmafia.preferences.Preferences;
+
+public class LocketManager {
+  private static final Set<Integer> knownMonsters = new TreeSet<>();
+  private static final Pattern REMINISCABLE_MONSTER = Pattern.compile("<option value=\"(\\d+)\"");
+
+  private static void addFoughtMonster(int monsterId) {
+    // Add monster id to pref ensuring distinct
+    Preferences.setString(
+        "_locketMonsters",
+        Stream.concat(
+                Arrays.stream(Preferences.getString("_locketMonsters").split(","))
+                    .filter(Predicate.not(String::isBlank)),
+                Stream.of(Integer.toString(monsterId)))
+            .distinct()
+            .collect(Collectors.joining(",")));
+  }
+
+  private LocketManager() {}
+
+  public static void rememberMonster(int monsterId) {
+    knownMonsters.add(monsterId);
+  }
+
+  public static Set<Integer> getRememberedMonsters() {
+    return Collections.unmodifiableSet(knownMonsters);
+  }
+
+  public static boolean remembersMonster(int monsterId) {
+    return knownMonsters.contains(monsterId);
+  }
+
+  public static void parseMonsters(final String text) {
+    // Visiting the reminisce page is a source of truth
+    knownMonsters.clear();
+
+    var m = REMINISCABLE_MONSTER.matcher(text);
```

Comment:
Just a thought, we need to also include monsters you've already fought today as they leave the list.